specification/mcpp/resource-manager/Microsoft.CloudPartnerProgramMembership/preview/2023-02-09-preview/mcpp.json

```diff
@@ -0,0 +1,447 @@
+        },
+        "description": "Fetch list of CloudPartnerProgramMembership purchase details for a given subscription",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/MembershipDetailsListResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CloudPartnerProgram/membership/{resourceName}": {
+      "get": {
+        "tags": ["CloudPartnerProgramMembership"],
+        "operationId": "membership_GetPurchaseDetails",
+        "x-ms-examples": {
+          "getOperations": { "$ref": "./examples/getMcppPurchaseDetails.json" }
+        },
+        "description": "The operation to fetch purchase details of a CloudPartnerProgramMembership offer",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Purchase details of the CloudPartnerProgramMembership resource.",
+            "schema": {
+              "$ref": "#/definitions/MembershipDetailsResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": ["CloudPartnerProgramMembership"],
+        "operationId": "membership_CreatePurchase",
+        "x-ms-examples": {
+          "putOperations": { "$ref": "./examples/createMcppPurchase.json" }
+        },
+        "description": "Create a CloudPartnerProgramMembership purchase",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ProductCodeParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
We will likely need to support the 202 async pattern.... You can still keep the 200 as well in case you can in fact complete synchronously in the future, but I suspect at least initially you will need to do 202.

Reply:
ok, will make this change

Reply:
My bad - you can use 200/201 but then use provisioningState to indicate what state the resource is in. Internally, your user RP can update this state whenever your async process for fulfillment/charge has completed.

For more info, see: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#azure-asyncoperation-resource-format